src/multicall.provider.spec.ts

```diff
@@ -0,0 +1,235 @@
+import { createAddress, MockEthersProvider } from "./tests";
+import { MULTICALL2_ABI, MulticallContract, MulticallProvider } from "./multicall.provider";
+import { ContractCall } from "ethers-multicall";
+import { ethers } from "forta-agent";
+
+describe("MulticallProvider test suite", () => {
+  const MULTICALL_IFACE = new ethers.utils.Interface(MULTICALL2_ABI);
+
+  const mockEthersProvider: MockEthersProvider = new MockEthersProvider();
+  let mockMulticallProvider: MockEthersProvider;
+
+  const TEST_MULTICALL2_ADDRESSES: Record<number, string> = {
+    0: createAddress("0xffe"), // networkId and multicall2 address used for tests
+  };
+
+  const TEST_ABI = [
+    "function func1() external view returns (uint)",
+    "function func2() external view returns (uint)",
+    "function func3(uint) external view returns (uint)",
+    "function func4(address) external view returns (uint)",
+  ];
+  const TEST_IFACE = new ethers.utils.Interface(TEST_ABI);
+
+  const TEST_BLOCKS = [12, 13, 15, 22];
+
+  const TEST_ADDR = createAddress("0xa1");
+  const contract = new MulticallContract(TEST_ADDR, TEST_ABI);
+
+  const TEST_CALLS: ContractCall[] = [
+    contract.func1(),
+    contract.func2(),
+    contract.func3(ethers.BigNumber.from(5)),
+    contract.func4(createAddress("0xb1")),
+  ];
+
+  const TEST_OUTPUTS = TEST_CALLS.map((call) => ethers.BigNumber.from(ethers.utils.formatBytes32String(call.name)));
+
+  const addCallTo = (block: number | string, indexes: number[], inputs: any[]) => {
+    for (let index of indexes) {
+      mockMulticallProvider.addCallTo(TEST_ADDR, block, TEST_IFACE, TEST_CALLS[index].name, {
+        inputs: TEST_CALLS[index].params,
+        outputs: [TEST_OUTPUTS[index]],
+      });
+    }
+  };
+
+  const generateMockProviderCall = () => {
+    const _call = mockEthersProvider.call;
+
+    mockEthersProvider.call = jest.fn().mockImplementation(({ data, to, from }, blockTag) => {
+      if (to.toLowerCase() === TEST_MULTICALL2_ADDRESSES[0]) {
+        const args = MULTICALL_IFACE.decodeFunctionData("tryAggregate", data) as [
+          requireSuccess: boolean,
+          calls: Array<{
+            callData: string;
+            target: string;
+          }>
+        ];
+        const results = args[1].map((call) => {
+          const res = mockMulticallProvider.call({ data: call.callData, to: call.target }, blockTag);
+          if (!res && args[0]) throw Error("Call failed");
+          return { success: res ? true : false, returnData: res ? res : "0x" };
```

Comment:
```suggestion
          return { success: Boolean(res), returnData: res ? res : "0x" };
```
Something like this looks better for me in this particular case.